src/ResourceGenerator/StrategyInterface.php

```diff
@@ -23,6 +23,7 @@ public function createResource(
         $instance,
         Metadata\AbstractMetadata $metadata,
         ResourceGenerator $resourceGenerator,
-        ServerRequestInterface $request
+        ServerRequestInterface $request,
+        int $depth = 0
```

Comment:
Oof- this _is_ a BC break, as it changes a signature in an interface.

Reply:
yep :/ this was the thing i mentioned as i opened the PR. still don't know how to resolve this without breaking BC.

Not changing signature could end up in szenarios like this:
```php
$a = new stdClass;
$b = new stdClass;
$a->b = $b;
$b->a = $a;
// using custom strategy for `$b` without passing the depth param would cause the issue i try to solve here
```

Reply:
That was my point with one of my previous comments — this may be an excellent reason to break BC and have an immediate major release. As such, let's figure out what all we should change to make this as robust as possible for that release.

Reply:
sounds good to me. should i just create a new PR with a few ideas or did you want to discuss it in issues first?

Reply:
The main thing here is to ensure that this is the only change necessary to fix the depth issues.

If you know of other fixes that need to be done to address other issues, and which require BC breaks, submit those as separate patches.

In all likelihood, I won't do the major release immediately; I'm going to be looking at #42 as well, as I understand it represents a common Doctrine scenario, which means the component fails for it.